src/uv_tcp.c

```diff
@@ -48,11 +43,11 @@ void UvTcpMaybeFireCloseCb(struct UvTcp *t)
 
     assert(QUEUE_IS_EMPTY(&t->accepting));
     assert(QUEUE_IS_EMPTY(&t->connecting));
-
-    if (t->listener.data != NULL) {
+    if (!QUEUE_IS_EMPTY(&t->aborting)) {
         return;
     }
-    if (!QUEUE_IS_EMPTY(&t->aborting)) {
+
+    if (t->listeners) {
```

Comment:
Nit: we usually write this explicitly: ```if (t->listeners != NULL)```, just to help casual readers to know this is a pointer and not a counter or something else.

Reply:
Fixed this as well.